### DISCUSSION TITLE
How can I do an inverted selection of contains(), or and inverted filter()?

### DISCUSSION THREAD

**Discussion by MatiDiez** [cypress-io/cypress#15659] [Questions and Help] [answered]
This is probably very basic but I can't make it work. 

I have an element in a table I'm finding by saying: 
`cy.get('span').contains('Test')`

As part of my test scenario I make 3 copies of that so I end up having. 
Test: Name
Test: Name (1)
Test: Name (2)
Test: Name (3)

Now I need to Select all 4 of them, but Name is randomly generated in another scenario, so the way I have to identify them all is by saying: 
```
cy.get('span').contains('(1)')
cy.get('span').contains('(2)')
cy.get('span').contains('(3)')
**cy.get('span').contains('Test:')**
```

As it comes obvious, when I run the last one It focuses on any of the other 3. So, the only way I have to select the root one is by saying something like

`cy.get('span').not('contains', '(').contains('Test')`

But for some reason it is not working. How can I do an inverted selection of contains(), or and inverted filter()?

**Comment by jennifer-shehane** [comment, marked as answer]
Use `.not()` with the `:contains()` pseudo-selector for the inverse:

```js
cy.get('li').not(':contains("Archived")')        // items WITHOUT the text
cy.get('tr').filter(':not(.disabled)')           // inverse of a class filter
```

For a starting `cy.contains()` there’s no built-in negation, so select the set and `.not()`/`.filter()` it down.
